add preamble to 5.2 release notes

jlstrater opened the pull request:
<img width="689" alt="screenshot 2019-01-23 at 14 22 24" src="https://user-images.githubusercontent.com/4601577/51609464-7d1a5800-1f1a-11e9-9587-46fa1bb88430.png">

 .
 .
 .

<img width="699" alt="screenshot 2019-01-23 at 14 23 12" src="https://user-images.githubusercontent.com/4601577/51609471-81df0c00-1f1a-11e9-88f7-1c5b1fb520ad.png">


from: https://builds.gradle.org/repository/download/Gradle_Check_BuildDistributions/19377045:id/distributions/gradle-5.2-all.zip!/gradle-5.2-20190123130553%2B0000/docs/release-notes.html

eriwen commented:
@jlstrater we also ought to update `release-features.txt` for the RC. Could we include that here?

eriwen commented on subprojects/docs/src/docs/release/notes.md:
```diff
@@ -1,6 +1,8 @@
 The Gradle team is excited to announce Gradle 5.2.
 
-This release features [1](), [2](), ... [n](), and more.
+This release features the [Java Platform plugin](userguide/java_platform_plugin.html), [official support for additional Windows native toolchains](userguide/native_software.html#native-binaries:tool-chain-support), [service injection into project extensions](userguide/custom_plugins.html#service_injection), and more.
```
> Java Platform plugin

Can we say "... for [such and such purpose]" briefly?

> Service injection into project extensions

What the what?

jlstrater commented on subprojects/docs/src/docs/userguide/upgrading_version_5.adoc:
```diff
@@ -33,6 +33,13 @@ Some plugins will break with this new version of Gradle, for example because the
+
+=== Potential breaking changes
+
+none
```
Please add them here if this isn't true! (also we should update the preamble).

jlstrater commented:
@eriwen I moved the section for plugin authors down in the list, removed it from the preamble, and added the other two to release-features.txt 

I also added a section to the upgrade guide which says there are no breaking changes for this release. Please someone correct me if that is wrong.

@adammurdoch I cleaned up the section you wrote about service injection into project extensions. Please review for accuracy. I removed the TBDs since they seem like good candidates for the linked user guide section instead of the release notes. If so, can you please add them there in a separate PR?

big-guy commented on subprojects/docs/src/docs/release/notes.md:
```diff
@@ -1,6 +1,9 @@
 The Gradle team is excited to announce Gradle 5.2.
 
-This release features [1](), [2](), ... [n](), and more.
+This release features the [Java Platform plugin](userguide/java_platform_plugin.html) which allows users to declare and publish platforms including as BOMs, [official support for additional Windows native toolchains](userguide/native_software.html#native-binaries:tool-chain-support), and more.
```
I just added another blurb about the new C++ plugins, I think we should promote those here over the mingw/cygwin stuff.  I think those used to work before, but we just had no coverage for them really.

I would say something like...
> This release features the [Java Platform plugin](linky), which allows you to declare a platform.  Like Maven BOMs, platforms can be used to define a set of versions for dependencies that are known to work together. These versions can be published and consumed elsewhere as dependency recommendations.

@ljacomet does this sound about right? ☝️ Is the Java Platform plugin chapter referenced anywhere in the navigation sidebar?

and continue with...
> This release also has a new set of [C++ plugins](userguide/cpp_plugins.html) that replace the software model C++ plugins. You can go for a spin by running `gradle init` and following the instructions to generate a C++ project. We recommend that C++ projects use the `cpp-application` or `cpp-library` plugin instead of the `cpp` plugin when starting new projects.

ljacomet replied:
@big-guy That's a really good suggestion for platforms. 👍 from me.
The new plugin is not referenced in the navigation sidebar at all.
I am going top open a PR that will add a section similar to [this one](https://docs.gradle.org/nightly/userguide/building_java_projects.html#sec:building_java_libraries)

ljacomet replied:
In that case, I believe that dropping
> including as BOMs
from the sentence reads better. After all, once Gradle Metadata is around, this feature is more than BOMs.

jlstrater replied:
👍 I already changed it.

ljacomet replied:
PR for having the reference to platforms in build java projects: #8329

big-guy commented on subprojects/docs/src/docs/release/notes.md:
```diff
@@ -65,9 +54,18 @@ This release includes some improvements to Gradle's console integration on Windo
 ## Support for setting environment variables when using Gradle TestKit
   
-Gradle [TestKit](userguide/test_kit.html) based tests can now specify environment variables via [`GradleRunner`](javadoc/org/gradle/testkit/runner/GradleRunner.html). These environemnt variables will be visible to the build under test.
+Gradle [TestKit](userguide/test_kit.html) based tests can now specify environment variables via [`GradleRunner`](javadoc/org/gradle/testkit/runner/GradleRunner.html). These environment variables will be visible to the build under test.
```
Thanks for catching this!

big-guy commented on subprojects/docs/src/docs/release/release-features.txt:
```diff
@@ -0,0 +1,3 @@
+- the Java Platform plugin which allows users to declare and publish platforms including as BOMs
+- official support for additional Windows native toolchains
```
Like my suggestion above, maybe something like...

```
- Define sets of dependencies that work together with Java Platform plugin
- New C++ plugins with dependency management built-in
- New project types for init task
```

jlstrater commented on subprojects/docs/src/docs/release/notes.md:
```diff
@@ -36,13 +41,13 @@ See more information about the [Gradle native project](https://github.com/gradle
 ### Recommend C++ projects use new C++ plugins
 
-We introduced new C++ plugins [last year](https://blog.gradle.org/introducing-the-new-cpp-plugins) that we thought were more feature rich and familiar to Gradle users. At the time, we thought they weren't mature enough for serious adoption.
+We introduced [new C++ plugins](https://blog.gradle.org/introducing-the-new-cpp-plugins) last year that we thought were more feature rich and familiar to Gradle users.
```
I think this is the confusing part @big-guy We call them the "new" plugins, but they aren't really new in this release.
